### PR TITLE
Add API PaginatedTable component

### PR DESCRIPTION
**Pull request by laurakwhit**
## Description & motivation 💭 
Adds an API paginated option for the `PaginatedTable` Holocene component and refactors the `Schedules` table to use it.

### Screenshots (if applicable) 📸 
| Before | After |
|--|--|
|Screenshot 2024-08-09 at 1 47 27 PM|Screenshot 2024-08-09 at 1 50 40 PM|
|Screenshot 2024-08-09 at 1 47 04 PM|Screenshot 2024-08-09 at 1 46 28 PM|
|Screenshot 2024-08-09 at 1 47 17 PM|Screenshot 2024-08-09 at 1 46 46 PM|

### How was this tested 👻 
- [ ] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 
Verify the following states
   - empty
   - loading
   - error
   - no results from filter/search
   - multiple results

work as expected in the following tables
   - [ ] Workflows 
   - [ ] Schedules

### CODE REVIEW

File: src/lib/holocene/table/paginated-table/api-paginated.svelte [hunk 1]
```diff
@@ -0,0 +1,229 @@
+<script lang="ts">
```
Comment (GiantRobots) [inline src/lib/holocene/table/paginated-table/api-paginated.svelte:1]: The rename is gonna cause a BUNCH of tables in cloud to need re-imported when we update the component. No notes just saying. Compilation errors will help there

Comment (laurakwhit) [reply src/lib/holocene/table/paginated-table/api-paginated.svelte:1]: We're currently only using this table in one place which was updated [here](https://github.com/temporalio/ui/pull/2261/files#diff-9658934782fa9de1d3992e17315fd3516efec1cd7a6cc8da42aeebd19f9ca130R5), so imports have been covered ✔️

File: src/lib/holocene/table/paginated-table/api-paginated.svelte [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    total?: string | number;
+    pageSizeSelectLabel: string;
+    emptyStateMessage: string;
+    fallbackErrorMessage: string;
+    itemsKeyname?: string;
+    previousButtonLabel: string;
+    nextButtonLabel: string;
+  };
+
+  type PaginatedRequest<T> = (
```
Comment (rossedfort) [inline src/lib/holocene/table/paginated-table/api-paginated.svelte:36]: Can you put this in a `<script context="module" />` and export it? It makes it handy when using paginated tables.

File: src/lib/pages/schedules.svelte [hunk 3]
```diff
@@ -59,66 +47,69 @@
...
-        <SchedulesTableRow {schedule} />
-      {:else}
-        <TableRow>
-          <td class="hidden xl:table-cell" />
-          <td class="hidden xl:table-cell" />
-          <td colspan="3">
+    </div>
+
+    <tr slot="headers" class="text-left">
+      <th class="px-2">{translate('common.status')}</th>
```
Comment (rossedfort) [inline src/lib/pages/schedules.svelte:89]: We may want to consider adding this `px-2` into the PaginatedTable component styles

File: src/lib/holocene/table/paginated-table/api-paginated.svelte [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+      {#if $$slots.error}
+        <slot name="error" />
+      {:else}
+        <EmptyState title="">
+          <Alert
+            intent="error"
+            title={error?.message ?? fallbackErrorMessage}
+          />
+        </EmptyState>
+      {/if}
```
Comment (rossedfort) [inline src/lib/holocene/table/paginated-table/api-paginated.svelte:186]:
```suggestion
        <slot name="error" >
          <EmptyState title="">
            <Alert
              intent="error"
              title={error?.message ?? fallbackErrorMessage}
            />
          </EmptyState>
        </slot>
```

File: src/lib/holocene/table/paginated-table/api-paginated.svelte [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    {:else if $$slots.empty}
+      <slot name="empty" />
+    {:else}
+      <EmptyState title="" content={emptyStateMessage} />
+    {/if}
```
Comment (rossedfort) [inline src/lib/holocene/table/paginated-table/api-paginated.svelte:191]:
```suggestion
    <slot name="empty">
      <EmptyState title="" content={emptyStateMessage} />
    </slot>
```

**Review by rossedfort — COMMENTED** [review]
We can use the slot's fallback value instead of checking its presence

File: src/lib/holocene/table/paginated-table/api-paginated.svelte (hunk 5 above)
Comment (rossedfort) [reply src/lib/holocene/table/paginated-table/api-paginated.svelte:191]: Maybe we also want an optional `emptyStateTitle` prop?

File: src/lib/holocene/table/paginated-table/api-paginated.svelte (hunk 4 above)
Comment (rossedfort) [reply src/lib/holocene/table/paginated-table/api-paginated.svelte:186]: Maybe we also want an optional `errorTitle` prop?

**Review by rossedfort — APPROVED** [review]

**Pull request merged**